jobs/e2e-runner.groovy

```diff
@@ -0,0 +1,127 @@
+      }
+    }
+
+    if (isPR) {
+      concurrentBuild()
+      throttleConcurrentBuilds {
+        maxPerNode(defaults.maxBuildsPerNode)
+        maxTotal(defaults.maxTotalConcurrentBuilds)
+      }
+    }
+
+    logRotator {
+      daysToKeep defaults.daysToKeep
+    }
+
+    if (isPR) { // set up GitHubPullRequest build trigger
+      triggers {
+        pullRequest {
+          admin('deis-admin')
+          cron('H/5 * * * *')
+          useGitHubHooks()
+          triggerPhrase('OK to test')
+          orgWhitelist(['deis'])
+          allowMembersOfWhitelistedOrgsAsAdmin()
+          // this plugin will update PR status no matter what,
+          // so until we fix this, here are our default messages:
+          extensions {
+            commitStatus {
+              context('ci/jenkins/pr')
+              triggeredStatus("Triggering workflow-e2e/e2e-runner build...")
+              startedStatus("Starting workflow-e2e/e2e-runner build...")
+              completedStatus('SUCCESS', "Merge with caution! Test job(s) may still be in progress...")
+              completedStatus('FAILURE', 'Build/deploy returned failure(s).')
+              completedStatus('ERROR', 'Something went wrong.')
+            }
+          }
+        }
+      }
+    }
+
+    parameters {
+      stringParam('DOCKER_USERNAME', 'deisbot', 'Docker Hub account name')
+      stringParam('DOCKER_EMAIL', 'dummy-address@deis.com', 'Docker Hub email address')
+      stringParam('QUAY_USERNAME', 'deisci+jenkins', 'Quay account name')
+      stringParam('QUAY_EMAIL', 'deisci+jenkins@deis.com', 'Quay email address')
+      stringParam('sha1', 'master', 'Specific Git SHA to test')
+    }
+
+    triggers {
+      githubPush()
+    }
+
+    wrappers {
+      timestamps()
+      colorizeOutput 'xterm'
+      credentialsBinding {
+        string("DOCKER_PASSWORD", "0d1f268f-407d-4cd9-a3c2-0f9671df0104")
+        string("QUAY_PASSWORD", "c67dc0a1-c8c4-4568-a73d-53ad8530ceeb")
+        string("GITHUB_ACCESS_TOKEN", "8e11254f-44f3-4ddd-bf98-2cabcb7434cd")
+      }
+    }
+
+    steps {
+      dockerPush = isPR ? 'push-e2e-runner-immutable' : 'push-e2e-runner-mutable'
+      shell """
+        #!/usr/bin/env bash
+        set -eo pipefail
+
+        docker login -e="\$DOCKER_EMAIL" -u="\$DOCKER_USERNAME" -p="\$DOCKER_PASSWORD"
+        DEIS_REGISTRY='' make docker-build ${dockerPush}
+
+        docker login -e="\$QUAY_EMAIL" -u="\$QUAY_USERNAME" -p="\$QUAY_PASSWORD" quay.io
+        DEIS_REGISTRY=quay.io/ make docker-build ${dockerPush}
+
+        mkdir -p "/tmp/${JOB_NAME}~/${BUILD_NUMBER}
+        eval $(make image)
+        # if triggered by pull request plugin, use ghprbActualCommit
+        export ACTUAL_COMMIT="\${ghprbActualCommit}"
+        # if manually triggered, use sha1
+        if [ -z "\${ghprbActualCommit}" ]; then
+        	export ACTUAL_COMMIT="\${sha1}"
+        fi
+        echo ACTUAL_COMMIT="\${ACTUAL_COMMIT}" >> /tmp/${JOB_NAME}/${BUILD_NUMBER}/env.properties
```

Comment:
might consider a var in this groovy file to hold the `env.properties` filepath so it only needs updating in one place:
```
propertiesFilepath = "/tmp/${JOB_NAME}/${BUILD_NUMBER}/env.properties"
```
then can interpolate in shell script as usual: `${propertiesFilepath}` here and below in the `propertiesFile(propertiesFilepath)`